api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -51,11 +54,36 @@ public void bufferBillingProject() {
     billingProjectBufferEntryDao.save(entry);
   }
 
+  public void syncBillingProjectStatus() {
+    BillingProjectBufferEntry entry = billingProjectBufferEntryDao
```

Comment:
I have some concerns about degenerate situations here, e.g. the oldest project stuck in this state forever. Maybe grab the first N instead and iterate? (maybe 5)

There's another ticket for handling that situation, but we'll probably want to allow a generous timeout and we don't want all projects to be stuck in the meantime.

Reply:
Hmm, yeah that's a good point. I think it's a very valid concern given the instability of FC. How about adding another field to the model to track the last synced time? We can change the orderBy to use that field and the sync will give every entry a chance to be synced without being blocked by a previous failing entry.

This also tracks the project creation time duration for free. `SELECT (last_synced_time - creation_time) FROM e WHERE e.status=AVAILABLE, ASSIGNED, ASSIGNING`

This also lets us keep the pattern where each invocation of the sync does a small piece of work (n=1) and we can control its throughput by adjusting only the cron frequency. I prefer that because it's easier to adjust and understand throughput with just 1 variable and it distributes load more evenly.

Reply:
Sure - that sounds reasonable